enterprise/models.py

```diff
@@ -1912,6 +1912,16 @@ def __str__(self):
         """
         return "<EnterpriseCatalogQuery '{title}' >".format(title=self.title)
 
+    def delete(self, *args, **kwargs):
+        """
+        Deletes this ``EnterpriseCatalogQuery``.
```

Comment:
Didn't the query end up deleted in this case anyways? *Do* we want to explicitly delete it? A catalog query can be associated with another catalog. If someone deleted the catalog (and it didn't delete the query) we could just hook the existing query up to the new catalog. I'm not sure I see the utility in allowing deletion. Just seems like another way for someone to then accidentally the query now too lol

Reply:
From the ticket AC:

> Add an alert to detect if a catalog query has been deleted

which is why I didn't _**prevent**_ the delete so to speak. But I'm more than happy to prevent it. Agreed, the management command helps us rectify any deletes that would otherwise need to happen.

Reply:
I am dumb. I forgot delete is already inherent to the class and it *was* a query that was deleted. You're just adding an override that adds actual logging which is exactly what we want. 👍